### PR TITLE
API: Search Issues, dont show 500 if filter result in empty list

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by lunny** [comment]
But at least we need to know there is a bug there.

**Review by delvh — APPROVED** [review]
@lunny I have no idea why it should be an error that no issue matched the filter.
The combination of returning no error and `0` should be hint enough for that.

File: models/issue.go [hunk 1]
```diff
@@ -1543,7 +1543,7 @@ func CountIssues(opts *IssuesOptions) (int64, error) {
 		return 0, fmt.Errorf("Find: %v", err)
 	}
 	if len(countsSlice) < 1 {
```
Comment (delvh) [inline models/issue.go]: Alternatively, we could check for `len(countsSlice) == 0`.
Which do you prefer?

Comment (6543) [reply models/issue.go]: I think it does not matter - but I'll change it as you are right - this is strange

**Comment by 6543** [comment]
> But at least we need to know there is a bug there.

I would also consider a filter that is to restrictive and so return 0 not an error

File: models/issue.go (hunk 1 above)
Comment (6543) [inline models/issue.go]:
```suggestion
	if len(countsSlice) == 0 {
```

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — COMMENTED** [review]
Need to be improved in my mind.

1. the sql `select count(t1.id) from t1 inner join t2 on t1.id=t2.id` always return 1 rows with a column value `0` or `123`.
2. why the `Find` result becomes 0 rows, I think it's a bug caused by `opts.Page>0`, then there is a `limit 10,20` in the sql, or maybe something else related to `limit`. How it happens? Is it correct?
3. the `RepoID int64` in countsSlice struct seems useless (and incorrect .... there is no such column in result)

Since we are fixing bugs, I think if there is a chance, we should fix the root case.

**Comment by wxiaoguang** [comment]
I proposed a full fix:

https://github.com/wxiaoguang/gitea/commit/18aeedf8a784401659279525d5223f889f312cd5

**Comment by wxiaoguang** [comment]
I think we can have this fix in 1.16.6 since it's a bug.

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**